### PR TITLE
Implemented the AudioFeedback base class

### LINKED ISSUE #171: Partially ported Audio subpackage from neurolab-desktop
**Actual Behaviour**
* _io.neurolab.output.audio_ subpackage is partially ported and is incomplete without the essential UI based AudioFeedback class for providing the audio feedback of the program modes.

**Expected Behaviour**
* _io.neurolab.output.audio_ subpackage should be complete and should have all the essential classes which include -> AudioFeedback, FocusAudioFeedback and JSynThread class.
* This would help for the completion of the Neuro-Audio Feedback Module and therefore module level tests can be carried out afterward for further validation.

**Would you like to work on the issue?**
I would like to work on this issue.

### PR DESCRIPTION
**Pull request by suresh-gandhi**
Fixes #171.

**Changes**: 
* Implemented the AudioFeedback base class (belonging to the output.audio package) in Android version.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications are done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

**Comment by suresh-gandhi** [comment]
@CloudyPadmal @mariobehling Could you please review the PR. Let me know if something needs to be changed. Thanks.

**Comment by ShridharGoel** [comment]
There are some unused fields and other issues as pointed out by Codacy. Please try to fix them.

File: app/src/main/java/io/neurolab/main/output/audio/AudioFeedback.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    private Synthesizer synth;
+    private LineOut lineOut;
+    private FloatSample[] sample;
+    private VariableRateDataReader[] samplePlayer;
+    private int numSamples = 5;
+    private float[] volume = {.5f, .5f, .5f, .5f, .7f};
+    private float[] defaultVolume = {.5f, .5f, .5f, .5f, .7f};
+    private String[] soundMixSamples = {"audio/pad_.wav", "audio/pad1.wav", "audio/lownoise.wav", "audio/pad2.wav", "audio/forest.wav"};
+    private Bitmap feedbackMixImage;
+    private int windowSize = 400;
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/output/audio/AudioFeedback.java]: I think these parameters are much related to the UI for the Feedback. No need of these in the base class. We can make UI related changes in the Activity and Fragment @suresh-gandhi .

File: app/src/main/java/io/neurolab/main/output/audio/AudioFeedback.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            for (int i = 0; i < numSamples; i++) {
+                soundMixSamples[i] = this.config.getPref(Config.audiofeedback, String.valueOf(Config.audiofeedback_params.sample) + i);
+                volume[i] = Float.valueOf(config.getPref(Config.audiofeedback, String.valueOf(Config.audiofeedback_params.volume) + i));
+            }
+            int x = Integer.valueOf(config.getPref(Config.audiofeedback, String.valueOf(Config.audiofeedback_params.x)));
+            int y = Integer.valueOf(config.getPref(Config.audiofeedback, String.valueOf(Config.audiofeedback_params.y)));
+            currentMixPoint.set(x, y);
+        }
+
+        colorImage = BitmapFactory.decodeFile(ResourceManager.getInstance().getResource(context, "4colors.jpg").getAbsolutePath());
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/output/audio/AudioFeedback.java]: Same here. We can make UI related changes in the Activity/Fragment.

**Review by jddeep — CHANGES_REQUESTED** [review]

**Comment by jddeep** [comment]
@suresh-gandhi Just go through the whole class once and keep only functionality related changes here in the base class. Because that will help us when we design and develop the UI. 🚀

**Comment by suresh-gandhi** [comment]
@jddeep Thanks for your review. I removed android.graphics usages (to be reserved for UI Activity/Fragment class). However, volume, max volume and other variables are being loaded (their default values mostly) from the Config File via Resource Manager here and even getting modified at some places. Hence I have added getters for them to be directly accessed by the UI class to use them for display.

**Review by jddeep — APPROVED** [review]
@suresh-gandhi Good job 👍

**Comment by suresh-gandhi** [comment]
@CloudyPadmal @mariobehling Could you please review and merge the PR.

**Review by CloudyPadmal — APPROVED** [review]

**Pull request merged**